lib/ecto/query/inspect.ex

```diff
@@ -70,11 +70,11 @@ defimpl Inspect, for: Ecto.Query do
 
   defp bound_from(from, name), do: ["from #{name} in #{unbound_from from}"]
 
-  defp unbound_from(nil),           do: "query"
-  defp unbound_from({source, nil}), do: inspect source
-  defp unbound_from({nil, schema}),  do: inspect schema
-  defp unbound_from(from = {source, schema}) do
-    inspect if source == schema.__schema__(:source), do: schema, else: from
+  defp unbound_from(nil), do: "query"
+  defp unbound_from(%FromExpr{source: source, schema: nil}), do: inspect source
+  defp unbound_from(%FromExpr{source: nil, schema: schema}),  do: inspect schema
+  defp unbound_from(%FromExpr{source: source, schema: schema}) do
+    inspect if source == schema.__schema__(:source), do: schema, else: {source, schema}
```

Comment:
Outputting `{source, schema}` here, at it is still external representation of `from`